src/Workspaces/CSharp/Portable/LanguageServices/CSharpTypeInferenceService.TypeInferrer.cs

```diff
@@ -240,6 +241,8 @@ protected override IEnumerable<TypeInferenceInfo> InferTypesWorker_DoNotCallDire
                 }
             }
 
+            private IEnumerable<TypeInferenceInfo> InferTypeInAnonymousObjectCreation(AnonymousObjectCreationExpressionSyntax expression) => InferTypes(expression.SpanStart);
```

Comment:
1. you need to pass the token in and validate that it's the 'new' token.  For examlpe, we should not infer a type in "new {$$" or "new { }$$".

2. If you use=> then wrap, so that it is visible on github.